location/__init__.py

```diff
@@ -0,0 +1,62 @@
+import os
+import sys
+import json
+import re
+import requests
+
+sys.path.append("../")
+from utils import db, ms
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+if os.environ.get("GOOGLE_MAPS_KEY"):
+    key = os.environ.get("GOOGLE_MAPS_KEY")
+    REQUEST_URL = f"https://maps.googleapis.com/maps/api/geocode/json?key={key}"
+    REQUEST_URL += "&address={}"
+
+
+def get_location(query):
```

Comment:
Are you calling get_location once per document? Is it possible to do this in a batch-wise way where we pass in a bunch of objects and it returns a bunch of lon/lats (and updates mongo accordingly). That way you only use 1 mongo request as opposed to N mongo requests.

Reply:
It's not being called anywhere yet, I'm still trying to think what the best approach is.

Every article is going to have at least 1 mongo request (the Location.objects.get(name=....) to check if it is present in Mongo. I suppose we could use an in-memory structure to prevent this, but I definitely agree that only making one large "push" to Mongo is smart.